File: W1/W1D4/lib/merge_sort.rb
def merge_sort(array)
  if array.length <= 1
    return array
  end

  # if array.length == 2
  #   case array[0] <=> array[1]
  #     when -1
  #       return array
  #     when 0
  #       return array
  #     when 1
  #       return array.reverse
  #     end
  #   end

  mid = array.length/2
  left = array[0..mid]
  right = array[mid+1..-1]
  merge(merge_sort(left), merge_sort(right))
  #
  # sorted_left =  merge_sort(left)
  # sorted_right = merge_sort(right)
  #
  # [sorted_left] + [sorted_right]
  #
  # p "Sorted left = #{sorted_left}"
  # p "Sorted right = #{sorted_right}"
  # p array
  #
end

def merge(arr1, arr2)

end

File: W6/W6D1/merge_sort.js
var mergeSort = function(array){
  if (array.length <= 1){
    return array;
  } else {
    var midpoint = Math.round(array.length / 2);
    var left = array.slice(0, midpoint);
    var right = array.slice(midpoint, array.length);

    return merge(mergeSort(left), mergeSort(right));
  }
}

var merge = function(arr1, arr2){
  var merged = [];

  while(arr1.length > 0 && arr2.length > 0){
    if (arr1[0] < arr2[0]){
      merged.push(arr1.shift());
    } else {
      merged.push(arr2.shift());
    }
  }
  return merged.concat(arr1, arr2);
}